src/runtime/icd.h

```diff
@@ -113,6 +113,8 @@
 #define clCreateEventFromGLsyncKHR _clCreateEventFromGLsyncKHR
 #endif // OCLGRIND_ICD
 
+#include "core/common.h"
```

Comment:
The convention here is to forward-declare rather than including headers inside headers (to help keep compilation times down), so please just forward-declare the `Image` struct in the `oclgrind` namespace declaration around line 142 in this file instead.

Reply:
Yeah, the include was a bit dubious. Anyway, in order to forward-declare the Image struct I had to change it from a C-style `typedef` to a C++ struct (in the sense it is known simply by `Image` and not `struct Image` like in C).